core/modules/czar/Czar.cc

```diff
@@ -0,0 +1,328 @@
+            _clientToQuery.insert(std::make_pair(ctId, uq));
+            LOGF(_log, LOG_LVL_DEBUG, "Remembering query: (%s, %s) (new map size: %s)" %
+                 clientId % threadId % _clientToQuery.size());
+        }
+    }
+
+    // return all info to caller
+    result[1] = resultName;
+    result[2] = lockName;
+    result[3] = uq->getProxyOrderBy();
+    LOGF(_log, LOG_LVL_DEBUG, "returning result to proxy: %s" % util::printable(result));
+    return result;
+}
+
+std::string
+Czar::killQuery(std::string const& query, std::string const& clientId) {
+
+    LOGF(_log, LOG_LVL_INFO, "KILL query: '%s'" % query);
+    LOGF(_log, LOG_LVL_INFO, "client ID: '%s'" % clientId);
+
+    // the query can be one of:
+    //   "KILL QUERY NNN" - kills currently running query in thread NNN
+    //   "KILL CONNECTION NNN" - kills connection associated with thread NNN
+    //                           and all queries in that connection
+    //   "KILL NNN" - same as "KILL CONNECTION NNN"
+
+    int threadId = ::parseKillQuery(query);
+    LOGF(_log, LOG_LVL_INFO, "thread ID: %s" % threadId);
+    if (threadId < 0) {
+        return "Failed to parse query: " + query;
+    }
+
+    ClientThreadId ctId(clientId, threadId);
+    ccontrol::UserQuery::Ptr uq;
+
+    {
+        std::lock_guard<std::mutex> lock(_mutex);
+
+        // find it in the client map based in client/thread id
+        auto iter = _clientToQuery.find(ctId);
+        if (iter == _clientToQuery.end()) {
+            return "Unknown thread ID: " + query;
+        }
+        uq = iter->second.lock();
+    }
+
+    // assume this cannot fail or throw
+    LOGF(_log, LOG_LVL_INFO, "Killing query for thread: %s" % threadId);
+    if (uq) {
+        uq->kill();
+    }
+
+    return std::string();
+}
+
+}}} // namespace lsst::qserv::czar
+
+namespace {
+
+// read configuration file
+lsst::qserv::StringMap
+readConfig(std::string const& configPath) {
+    using boost::property_tree::ini_parser::read_ini;
+    using boost::property_tree::ptree;
+
+    // read it into a ptree
+    ptree pt;
+    read_ini(configPath, pt);
+
+    // flatten
+    lsst::qserv::StringMap config;
+    for (auto& sectionPair: pt) {
+        auto& section = sectionPair.first;
+        for (auto& itemPair: sectionPair.second) {
+            auto& item = itemPair.first;
+            auto& value = itemPair.second.data();
+            config.insert(std::make_pair(section + "." + item, value));
+        }
+    }
+
+    return config;
+}
+
+int
+parseKillQuery(std::string const& aQuery) {
+    // the query that proxy passes us is all uppercase and spaces compressed
+    // but it may have trailing space which we strip first
+    std::string query = aQuery;
+    auto pos = query.find_last_not_of(' ');
+    query.erase(pos+1);
+
+    // try to match against one or another form of KILL
+    static const std::string prefixes[] = {"KILL QUERY ", "KILL CONNECTION ", "KILL "};
+    for (auto& prefix: prefixes) {
```

Comment:
Ah right -- since the inits are char* -- yup!  I'd bet they'd be small, on the stack, and that compiler would have the underlying char*s as static, tho', so I wouldn't be concerned...  But your way is explicit, readable, and compact as well so I'm fine with it. :-)